foreign_cc/extensions.bzl

```diff
@@ -23,22 +23,31 @@ def _init(module_ctx):
         register_built_pkgconfig_toolchain = True,
     )
 
-    versions = {
-        "cmake": _DEFAULT_CMAKE_VERSION,
-        "ninja": _DEFAULT_NINJA_VERSION,
-    }
+    cmake_version = _DEFAULT_CMAKE_VERSION
+    ninja_version = _DEFAULT_NINJA_VERSION
 
+    # Traverse all modules starting from the root one (the first in
+    # module_ctx.modules). The first occurrence of cmake or ninja tag wins.
+    # Multiple versions requested from the same module are rejected.
     for mod in module_ctx.modules:
-        if not mod.is_root:
-            for toolchain in mod.tags.cmake:
-                versions["cmake"] = toolchain.version
+        cmake_versions_count = len(mod.tags.cmake)
+        if cmake_versions_count == 1:
+            cmake_version = mod.tags.cmake[0].version
+            break
+        elif cmake_versions_count > 1:
+            fail("More than one cmake version requested: {}".format(mod.tags.cmake))
```

Comment:
I think we should be able to support registering more than one version of cmake in a build. We should probably generate a version specific constraint that we apply to the toolchain targets so that a user can select a version constraint that way if they desire.

Reply:
We do support declaring more than one version of cmake, but each such declaration should be in a different module. Here we simply guard against multiple declarations in the same module.

Reply:
By removing the version number from the generated names though you will get duplicate definitions if a module defines one version and then the root requests a different version. This is also the rationale for restricting definitions to the root module at least for the time being.

Reply:
While we do allow specifying several versions of cmake/ninja, only one version wins—the one closest to the root module (see `break` in the "then" clause). Also, there can only be one version per module, so this should be very close to the old logic with the exception that the user can actually configure the versions.